### PR TITLE
PianoBot

### PR DESCRIPTION
**Pull request by ThatDevHuddy**
## Describe your project in 1-3 sentences
My team of 4 will build a self playing piano. We will use a teensy board to control solenoids to push the keys

## Hack Club Slack handle
@ Hudson Etkin

## Checklist before submitting proposal:

- [X ] I am a current high school student (or younger).
- [X ] I have submitted an RSVP at <https://hackclub.com/winter>
- [X ] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [ X] I have self-reviewed my project proposal.

### CODE REVIEW

**Comment by KaiEtkin** [comment]
My name is Kai and I am one of the members of this project.

[X ] I am a current high school student (or younger).
[X ] I have submitted an RSVP at https://hackclub.com/winter
[X ] I am a member of the Hack Club Slack (Join at https://hackclub.com/slack)
[ X] I have self-reviewed my project proposal.

Slack handle: @Kai Etkin

**Comment by dhruvaron** [comment]
My name is Dhruv and I am one of the members of this project.

[X ] I am a current high school student (or younger).
[X ] I have submitted an RSVP at https://hackclub.com/winter
[X ] I am a member of the Hack Club Slack (Join at https://hackclub.com/slack)
[ X] I have self-reviewed my project proposal.

Slack handle: @ Dhruv Aron

**Comment by TakumiJoeBritt** [comment]
* [x]  I am a current high school student (or younger).

* [x]  I have submitted an RSVP at https://hackclub.com/winter

* [x]  I am a member of the Hack Club Slack (Join at https://hackclub.com/slack)

* [x]  I have self-reviewed my project proposal.

My name is TJ and I am a member of this project.
My slack handle is takumi britt.

**Comment by bellesea** [comment]
heyy thanks so much for making a PR! we're back on tuesday and will start reviewing PRs in the order that they came on. 

to make the process smoother, please ensure you double checked that your file is properly formatted, you have all the parts you need, and that your plan is detailed enough for us to imagine how you're using the components in your budget to build your project and visualise you work on it :)

and fyi, we're looking for projects where/that:
- you can spend at least 10 days building (from Feb 14th to 23rd)
- you are able to modify or make your own in some ways
- you can learn something new from
- involves working with electronics or software on a lower level than a pre-assembled kit of parts

and, we normally don’t approve projects where you buy consumer components w/o modification like keyboards (buying PCB, keycaps, switches, and assembling it), PCs (buying motherboard, cpu, memory and plugging it together), and kits (eg. closed-sourced consumer drone kits). 

_(note: from a first glance of your project it looks SUPER COOL i'm wondering: are you following any sort of guides for this? can you please link the actual relevant amazon/etc. links? thank you!)_

our Jan 15 11:59pm ET deadline is for PRs to come in and not for final approval so we'll work with you to either accept this project OR give you a chance to submit another PR. 

if you have any questions, please do not hesitate to ask her or DM me (i'm belle on slack), thank you!

**Comment by ThatDevHuddy** [comment]
> heyy thanks so much for making a PR! we're back on tuesday and will start reviewing PRs in the order that they came on.
> 
> to make the process smoother, please ensure you double checked that your file is properly formatted, you have all the parts you need, and that your plan is detailed enough for us to imagine how you're using the components in your budget to build your project and visualise you work on it :)
> 
> and fyi, we're looking for projects where/that:
> 
> * you can spend at least 10 days building (from Feb 14th to 23rd)
> * you are able to modify or make your own in some ways
> * you can learn something new from
> * involves working with electronics or software on a lower level than a pre-assembled kit of parts
> 
> and, we normally don’t approve projects where you buy consumer components w/o modification like keyboards (buying PCB, keycaps, switches, and assembling it), PCs (buying motherboard, cpu, memory and plugging it together), and kits (eg. closed-sourced consumer drone kits).
> 
> _(note: from a first glance of your project it looks SUPER COOL i'm wondering: are you following any sort of guides for this? can you please link the actual relevant amazon/etc. links? thank you!)_
> 
> our Jan 15 11:59pm ET deadline is for PRs to come in and not for final approval so we'll work with you to either accept this project OR give you a chance to submit another PR.
> 
> if you have any questions, please do not hesitate to ask her or DM me (i'm belle on slack), thank you!

We are not following any guides. 
This doc has the links to all the parts https://docs.google.com/document/d/1pCTGWZPjfgzufPHXTuUPvIJxoMRuIZJ-XZey157PIz8/edit?usp=sharing

**Comment by bellesea** [comment]
@ThatDevHuddy cool! can you please update that in the file? also, if you're not referencing any guides of sorts, how do you know this will work? apologies for my lack of knowledge in this area...

**Comment by ThatDevHuddy** [comment]
There are many videos of successful player pianos using our approach on youtube, such as videos posted by this channel https://www.youtube.com/@japhillips87/videos or this video https://www.youtube.com/watch?v=K9tIqUlIAjU or this video https://www.youtube.com/watch?v=iLiZG25gSIg. I will be sure to update it in the file

**Comment by bellesea** [comment]
@ThatDevHuddy cool, thank you! we've started reviewing projects but due to the flurry last weekend (100+ PRs ksdjhfkjdshf) it might take a while before we review yours, thank you for your patience!

File: thatdevhuddy.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+Hook up solenoids and solenoid drivers in place of LEDs
+Build frame on piano
+3d print mounts for all parts
+Line up parts and mounts on frame and attach
+Test and tweak.
+
+
+## Budget
+
+What materials will you need for your project? Where will you get them? How much does it cost? Please include all materials, including components you already own. Make sure to factor in shipping costs and sales tax.
```
Comment (exu3) [inline thatdevhuddy.md:25]:
```suggestion
```

File: thatdevhuddy.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+All of the links to the parts are here: https://docs.google.com/document/d/1pCTGWZPjfgzufPHXTuUPvIJxoMRuIZJ-XZey157PIz8/edit?usp=sharing
+
+| Product           | Supplier/Link                         | Cost    |
+| ---------------   | ------------------------------------- | ------  |
+| MOSFET Drivers    | Amazon                                | $42.00  |
```
Comment (exu3) [inline thatdevhuddy.md:31]: Could you link to the product on Amazon? And do the same for the other line items on your budget please!

File: thatdevhuddy.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# PianoBot
+
+## Summary
+
+We are going to build a self-playing piano. Our team has 4 members. One of our group member’s has a brother who is very skilled at playing the piano. Although we do not possess the same talent as he does, we are determined to mimic his abilities through innovation and technology. Our design consists of solenoids placed on top of every key (we’re only doing 2 octaves to start) that is controlled by a Teensy that reads a MIDI roll and pushes the solenoids accordingly. We are not following any guides.
+
+## Plan
+
+Prototype Circuit with LEDs
+Write code to read MIDI roll and turn on and off LEDs accordingly
```
Comment (exu3) [inline thatdevhuddy.md:15]: What might this involve? Could you specify programming languages/frameworks/etc?

File: thatdevhuddy.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
+name: "@thatdevhuddy"
```
Comment (exu3) [inline thatdevhuddy.md:2]: Could you add the names of your teammates here? 

Format: `name: ["@wahoo", "@hooray", "@etc"]`

File: thatdevhuddy.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+We are going to build a self-playing piano. Our team has 4 members. One of our group member’s has a brother who is very skilled at playing the piano. Although we do not possess the same talent as he does, we are determined to mimic his abilities through innovation and technology. Our design consists of solenoids placed on top of every key (we’re only doing 2 octaves to start) that is controlled by a Teensy that reads a MIDI roll and pushes the solenoids accordingly. We are not following any guides.
+
+## Plan
+
+Prototype Circuit with LEDs
+Write code to read MIDI roll and turn on and off LEDs accordingly
+Hook up solenoids and solenoid drivers in place of LEDs
+Build frame on piano
+3d print mounts for all parts
```
Comment (exu3) [inline thatdevhuddy.md:18]: Are you going to design your own 3D models? If so, could you elaborate on your process for that? What CAD software might you use / do you have experience?

If not, could you link the 3D models you'll be using?

File: thatdevhuddy.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+## Summary
+
+We are going to build a self-playing piano. Our team has 4 members. One of our group member’s has a brother who is very skilled at playing the piano. Although we do not possess the same talent as he does, we are determined to mimic his abilities through innovation and technology. Our design consists of solenoids placed on top of every key (we’re only doing 2 octaves to start) that is controlled by a Teensy that reads a MIDI roll and pushes the solenoids accordingly. We are not following any guides.
+
+## Plan
+
+Prototype Circuit with LEDs
+Write code to read MIDI roll and turn on and off LEDs accordingly
+Hook up solenoids and solenoid drivers in place of LEDs
+Build frame on piano
```
Comment (exu3) [inline thatdevhuddy.md:17]: I assume you already have access to a piano you all can meet up and work on. Is this assumption correct?

**Review by exu3 — CHANGES_REQUESTED** [review]
This project sounds cool! I left some questions below.

File: thatdevhuddy.md (hunk 2 above)
Comment (ThatDevHuddy) [reply thatdevhuddy.md:31]: It's all in the doc I linked.

Comment (exu3) [reply thatdevhuddy.md:30]: Could you link it here as well?

File: thatdevhuddy.md (hunk 3 above)
Comment (exu3) [reply thatdevhuddy.md:15]: @ThatDevHuddy: I see that you have marked this comment as resolved; were you able to update your file with these details?

**Comment by exu3** [comment]
Please re-request review whenever you're ready!

**Review by ThatDevHuddy — COMMENTED** [review]
Is this what I was meant to do to request review?

**Comment by exu3** [comment]
Screenshot 2023-01-19 at 10 54 46 PM

Ah yes, I meant this button. I couldn't tell if you were ready for another review.

File: thatdevhuddy.md (hunk 5 above)
Comment (exu3) [reply thatdevhuddy.md:18]: Can you update your file with these details?

**Review by exu3 — APPROVED** [review]
Everything else looks good—could you just make sure to add these details: https://github.com/hackclub/winter/pull/355/files#r1082089935?

Once you do so, the @hackclub/bank-ops team will review and reach out on Slack with next steps.

**Review by MelSmith104 — COMMENTED** [review]
hey everyone! wow! this project looks like so much fun :) im so excited to see the final product! im dm'ing you now to get the process started for the grant!!

**Review by MelSmith104 — APPROVED** [review]

**Pull request merged**